pkg/ztunnel/xds/stream_processor.go

```diff
@@ -190,11 +217,29 @@ func (es *EndpointSource) SubscribeToEndpointEvents(ctx context.Context, wg *syn
 	for e := range newEvents {
 		switch e.Kind {
 		case resource.Upsert:
+			txn := es.sp.db.ReadTxn()
+			_, _, found := es.sp.enrolledNamespaceTable.Get(txn, table.EnrolledNamespacesNameIndex.Query(e.Object.GetNamespace()))
+			if !found {
+				es.sp.log.Debug("Skipping processing of CiliumEndpoint in unenrolled namespace",
+					logfields.K8sNamespace, e.Object.GetNamespace(),
+					logfields.Name, e.Object.GetName(),
+				)
+				break
+			}
```

Comment:
I think this can be done outside the switch statement to avoid duplication? Same for handleCESUpsert/handleCESDelete I believe.